FirstCommitForChecking

DenisKachan opened the pull request.

DenisKachan committed:
FirstCommitForChecking

VladimirVolkovVD commented on src/test/java/UI/BrowserConfig/SelenideConfiguration.java:
```diff
@@ -0,0 +1,34 @@
+    public static void setUpBasicConfigure() {
+        PropertyReader propertyReader = new PropertyReader();
+        Configuration.baseUrl = propertyReader.getProperty("baseUrl");
+        Configuration.headless = false;
```
переменные в проперти файл, если уже начал

VladimirVolkovVD commented on src/test/java/UI/Utils/TestConstants.java:
```diff
@@ -0,0 +1,32 @@
+public interface TestConstants {
+    Faker faker = new Faker();
+    String VALID_EMAIL = "den.kachan.98@mail.ru";
+    String VALID_PASSWORD = "proEVOlution20jomaQ";
```
Базовые переменные лучше в проперти (email, password)

VladimirVolkovVD commented on src/test/java/UI/Utils/TestConstants.java:
```diff
@@ -0,0 +1,32 @@
+    String VALID_EMAIL = "den.kachan.98@mail.ru";
+    String VALID_PASSWORD = "proEVOlution20jomaQ";
+    String INVALID_EMAIL = "vasia@55mail.ru";
+    String INVALID_PASSWORD = "adfdfsadf";
```
invalid data сгенерировать, нет смысла их хранить

VladimirVolkovVD commented on src/test/java/UI/Pages/CreateNewTestRunPage.java:
```diff
@@ -0,0 +1,35 @@
+    public SelenideElement doneButton = $(By.id("select-cases-done-button"));
+    public SelenideElement saveRunButton = $(By.id("save-run-button"));
+
+    public ListOfTestRunsPage successfulTestRunCreation() {
```
мараметр имени для тест рана лучше передать как параметр, и если он null взять рандомный валидный

VladimirVolkovVD commented on src/test/java/UI/Pages/CreateNewTestRunPage.java:
```diff
@@ -0,0 +1,35 @@
+        addCasesButton.click();
+        selectTestCaseCheckBox.click();
+        doneButton.click();
+        saveRunButton.click();
```
как узнать имя созданного тайтла? сли необходимо вернуть его из метода

VladimirVolkovVD commented on src/test/java/UI/Pages/LoginPage.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    public LoginPage loginWithInvalidData() {
+        login(INVALID_EMAIL, INVALID_PASSWORD);
```
не валидные данные можно сгенирировать

VladimirVolkovVD commented on src/test/java/UI/Steps/CaseSteps.java:
```diff
@@ -0,0 +1,26 @@
+        repositoryPage.clickCreateCaseButton();
+        createNewCasePage.createNewCase();
+        createNewCasePage.save();
+        repositoryPage.testCaseTitle.should(Condition.have(Condition.text(TEST_CASE_TITLE)));
```
почем проверка тут и какой тайтл пмы проверяем? откуда мы его указываем?

VladimirVolkovVD commented:
Тестовые данные зашиты в логику пейджей, нет возможности передавать их с уровня тестов

DenisKachan committed:
Commit with requested changes, '@step' annotations for allure reporting, logs:
1. All invalid data was removed from Constants file and set randomly in Test methods.
2. All browser configurations and valid email/password are written in Properties file.
3. All data (that is used and checked in tests) was removed from Pages and Steps level and was established on Test level.
4. Addition of step annotations with parameters for allure reporting and logs on LoginPage,LoginSignOutSteps (Just to be sure if i'm right).

DenisKachan committed:
Commit with following changes:
1.RetryAnalyzer added.
2.TestListener with screenshots in case of failer/skipped status added.
3.Smoke test file with ability of parallel run added.
